subsys/testsuite/include/zephyr/tc_util.h

```diff
@@ -121,14 +121,46 @@ static inline void test_time_ms(void)
 	} while (0)
 #endif
 
+static inline void print_nothing(const char *fmt, ...)
+{
+	ARG_UNUSED(fmt);
+}
+
 #ifndef TC_PRINT
+/* Need to check for CONFIG_ZTEST_NEW_API since the TC_PRINT
+ * is also used by the old ztest.
+ */
+#ifdef CONFIG_ZTEST_NEW_API
+#if defined(CONFIG_ZTEST_VERBOSE_OUTPUT)
+#define TC_PRINT(fmt, ...) PRINT_DATA(fmt, ##__VA_ARGS__)
+#else
```

Comment:
These configs are getting a bit deep. Can we add comments at each `endif` like:

`#endif /* CONFIG_ZTEST_NEW_API */`

Reply:
done.

Reply:
Apologies, I think I wasn't clear enough in my prior request:

Could we other `#endif /* CONFIG_{SOMETHING}` ? I didn't just mean `CONFIG_ZTEST_NEW_API` but the others as well.

Reply:
As you wish. :)

Reply:
As you wish. :)